cli/running/common.go

```diff
@@ -0,0 +1,206 @@
+	}
+
+	if projectCtx.DataDir == "" {
+		projectCtx.DataDir = filepath.Join(curDir, defaultLocalDataDir)
+	}
+	if projectCtx.DataDir, err = filepath.Abs(projectCtx.DataDir); err != nil {
+		return fmt.Errorf("Failed to get data dir absolute path: %s", err)
+	}
+
+	if projectCtx.LogDir == "" {
+		projectCtx.LogDir = filepath.Join(curDir, defaultLocalLogDir)
+	}
+	if projectCtx.LogDir, err = filepath.Abs(projectCtx.LogDir); err != nil {
+		return fmt.Errorf("Failed to get log dir absolute path: %s", err)
+	}
+
+	if projectCtx.ConfPath == "" {
+		projectCtx.ConfPath = filepath.Join(curDir, defaultLocalConfPath)
+	}
+	if projectCtx.ConfPath, err = filepath.Abs(projectCtx.ConfPath); err != nil {
+		return fmt.Errorf("Failed to get conf path absolute path: %s", err)
+	}
+
+	return nil
+}
+
+func collectInstancesFromConf(projectCtx *project.ProjectCtx) ([]string, error) {
+	var instances []string
+
+	// collect conf files
+	var confFilePaths []string
+	if fileInfo, err := os.Stat(projectCtx.ConfPath); err != nil {
+		return nil, fmt.Errorf("Failed to use conf path: %s", err)
+	} else if fileInfo.IsDir() {
+		for _, pattern := range confFilePatterns {
+			paths, err := filepath.Glob(filepath.Join(projectCtx.ConfPath, pattern))
+			if err != nil {
+				return nil, err
+			}
+
+			confFilePaths = append(confFilePaths, paths...)
+		}
+	} else {
+		confFilePaths = append(confFilePaths, projectCtx.ConfPath)
+	}
+
+	addedInstances := make(map[string]struct{})
+
+	// read files
+	for _, confFilePath := range confFilePaths {
+		conf, err := common.GetFileContent(confFilePath)
+		if err != nil {
+			return nil, fmt.Errorf("Failed to read configuration from file: %s", err)
+		}
+
+		instancesMap := make(map[string]interface{})
+		if err := yaml.Unmarshal([]byte(conf), instancesMap); err != nil {
+			return nil, fmt.Errorf("Failed to parse %s: %s", confFilePath, err)
+		}
+
+		for instanceID := range instancesMap {
+			instanceIDParts := strings.SplitN(instanceID, ".", 2)
+			if len(instanceIDParts) < 2 {
+				continue
+			}
+
+			appName := instanceIDParts[0]
+			if appName != projectCtx.Name {
+				continue
+			}
+
+			instanceName := instanceIDParts[1]
+			if _, found := addedInstances[instanceName]; found {
+				return nil, fmt.Errorf("Duplicate config section %s", instanceName)
+			}
+
+			instances = append(instances, instanceName)
+			addedInstances[instanceName] = struct{}{}
+		}
+	}
+
+	return instances, nil
+}
+
+func collectProcesses(projectCtx *project.ProjectCtx) (*ProcessesSet, error) {
+	processes := ProcessesSet{}
+
+	if projectCtx.WithStateboard {
+		process := NewStateboardProcess(projectCtx)
+		processes.Add(process)
+	}
+
+	if !projectCtx.StateboardOnly {
+		for _, instance := range projectCtx.Instances {
+			process := NewInstanceProcess(projectCtx, instance)
+			processes.Add(process)
+		}
+	}
+
+	return &processes, nil
+}
+
+func formatEnv(key, value string) string {
+	return fmt.Sprintf("%s=%s", key, value)
```

Comment:
I don't sure that it's really needed but such approach doesn't support several words values?
E.g. `KEY=VALUE1 VALUE2` is in fact `KEY=VALUE1`, should be `KEY="VALUE1 VALUE2"`

Reply:
I've just checked it - everything is OK with passing multi-words values. Using `KEY="VALUE1 VALUE2"` leads to setting `KEY` equal to `'"VALUE1 VALUE2"'` (with quotes).

Reply:
Well, could you add a test case for it?

Reply:
Decided that we shouldn't test `os/exec`, but if someone add quotes here, tests will fail.